openmls/src/group/public_group/diff/apply_proposals.rs

```diff
@@ -140,6 +142,15 @@ impl<'a> PublicGroupDiff<'a> {
             })
             .collect();
 
+        // apply group context extension proposal
+        let extensions = proposal_queue
+            .filtered_by_type(ProposalType::GroupContextExtensions)
+            .map(|queued_proposal| match queued_proposal.proposal() {
+                Proposal::GroupContextExtensions(extensions) => extensions.extensions().clone(),
+                _ => unreachable!(),
```

Comment:
Is there a reason why you're not using `filter_map` and an `if let` construction here? It would avoid the `unreachable`.